src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -23,6 +24,7 @@ public HttpParser() : this(showErrorDetails: true)
         public HttpParser(bool showErrorDetails)
         {
             _showErrorDetails = showErrorDetails;
+            _singleLineFeedEol = AppContext.TryGetSwitch("Microsoft.AspNetCore.Server.Kestrel.AcceptLineFeedAsLineTerminator", out var enabled) && enabled;
```

Comment:
See https://github.com/dotnet/aspnetcore/pull/41098/files#diff-35b875df5d58f4a5f4ace9648f3a2054b86bf054aa8c93128898f086ede4e47d for an example of adding switches.